src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1585,16 +1585,37 @@ public override BoundNode VisitCall(BoundCall node)
             // (Compare with CSharpOperationFactory.CreateBoundCallOperation.)
             if (!node.HasErrors)
             {
+                var savedState = this.State.Clone();
                 ImmutableArray<RefKind> refKindsOpt = node.ArgumentRefKindsOpt;
                 ImmutableArray<BoundExpression> arguments = RemoveArgumentConversions(node.Arguments, refKindsOpt);
                 ImmutableArray<int> argsToParamsOpt = node.ArgsToParamsOpt;
-                ImmutableArray<Result> results = VisitArgumentsEvaluate(arguments, refKindsOpt, method.Parameters, argsToParamsOpt, node.Expanded);
+                // We do a first pass to work through the arguments without making any assumptions
+                ImmutableArray<Result> results = VisitArgumentsEvaluate(arguments, refKindsOpt);
 
                 if (method.IsGenericMethod && HasImplicitTypeArguments(node))
```

Comment:
>method.IsGenericMethod [](start = 20, length = 22)

Not sure how this handles the case of a generic type `G<T>` with a non-generic method `M`, where we have an invocation expression `e.M(args)` but we have to recompute the type of the receiver `e` during this pass. Don't we have to recompute the method `M` as a member of the recomputed type of the receiver?

Reply:
Ah, I see a prototype comment for this case above.

---
In reply to: [191553261](https://github.com/dotnet/roslyn/pull/26828#discussion_r191553261) [](ancestors = 191553261)